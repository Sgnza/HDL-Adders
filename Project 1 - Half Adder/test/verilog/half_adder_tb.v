`timescale 1ps/1ps
`include "../../src/verilog/half_adder.v"
module half_adder_tb;

reg a_tb;
reg b_tb;
wire sum_tb;
wire cout_tb;

half_adder ha0(
    .a(a_tb),
    .b(b_tb),
    .sum(sum_tb),
    .cout(cout_tb)
);

task verify;
    input a_task;
    input b_task;
    input sum_task;
    input cout_task;

    begin
        a_tb = a_task;
        b_tb = b_task;
        #1;
        if(sum_task == sum_tb && cout_task == cout_tb) begin
            $display("CORRECTO: Dato de entrada a: %b, Dato de entrada b: %b, Dato de salida s esperado: %b, Dato de salida s obtenido: %b, Dato de salida c esperado: %b, Dato de salida c obtenido: %b", a_tb, b_tb, sum_task, sum_tb, cout_task, cout_tb);
        end else begin
            $display("INCORRECTO: Dato de entrada a: %b, Dato de entrada b: %b, Dato de salida s esperado: %b, Dato de salida s obtenido: %b, Dato de salida c esperado: %b, Dato de salida c obtenido: %b", a_tb, b_tb, sum_task, sum_tb, cout_task, cout_tb);

        end
    end

endtask

initial begin
  $dumpfile("half_adder_tb.vcd");
  $dumpvars(0, half_adder_tb);
end

initial begin

verify(
    1'b0,
    1'b0,
    1'b0,
    1'b0
);

verify(
    1'b1,
    1'b0,
    1'b1,
    1'b0
);

verify(
    1'b0,
    1'b1,
    1'b1,
    1'b0
);

verify(
    1'b1,
    1'b1,
    1'b0,
    1'b1
);

end

endmodule