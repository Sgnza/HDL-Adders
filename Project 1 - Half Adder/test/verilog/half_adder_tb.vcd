$date
	Fri Jan 12 23:25:09 2024
$end
$version
	Icarus Verilog
$end
$timescale
	1ps
$end
$scope module half_adder_tb $end
$var wire 1 ! s_tb $end
$var wire 1 " c_tb $end
$var reg 1 # a_tb $end
$var reg 1 $ b_tb $end
$scope module hf0 $end
$var wire 1 # a $end
$var wire 1 $ b $end
$var wire 1 ! s $end
$var wire 1 " c $end
$upscope $end
$scope task verify $end
$var reg 1 % a_task $end
$var reg 1 & b_task $end
$var reg 1 ' c_task $end
$var reg 1 ( s_task $end
$upscope $end
$upscope $end
$enddefinitions $end
$comment Show the parameter values. $end
$dumpall
$end
#0
$dumpvars
0(
0'
0&
0%
0$
0#
0"
0!
$end
#1
1!
1#
1(
1%
#2
1$
0#
1&
0%
#3
0!
1"
1#
1'
0(
1%
#4
