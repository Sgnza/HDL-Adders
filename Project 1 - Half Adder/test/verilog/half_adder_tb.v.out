#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "C:\iverilog\lib\ivl\system.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "C:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "C:\iverilog\lib\ivl\va_math.vpi";
S_0000022a6485ada0 .scope module, "half_adder_tb" "half_adder_tb" 2 3;
 .timescale -12 -12;
v0000022a64872920_0 .var "a_tb", 0 0;
v0000022a648a40b0_0 .var "b_tb", 0 0;
v0000022a648a4150_0 .net "c_tb", 0 0, L_0000022a648a4290;  1 drivers
v0000022a648a41f0_0 .net "s_tb", 0 0, L_0000022a648a67e0;  1 drivers
S_0000022a6489b770 .scope module, "hf0" "half_adder" 2 10, 3 1 0, S_0000022a6485ada0;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /OUTPUT 1 "s";
    .port_info 3 /OUTPUT 1 "c";
L_0000022a648a67e0 .functor XOR 1, v0000022a64872920_0, v0000022a648a40b0_0, C4<0>, C4<0>;
v0000022a64872db0_0 .net "a", 0 0, v0000022a64872920_0;  1 drivers
v0000022a6489b310_0 .net "b", 0 0, v0000022a648a40b0_0;  1 drivers
v0000022a6489b3b0_0 .net "c", 0 0, L_0000022a648a4290;  alias, 1 drivers
v0000022a6489b450_0 .net "s", 0 0, L_0000022a648a67e0;  alias, 1 drivers
L_0000022a648a4290 .arith/mult 1, v0000022a64872920_0, v0000022a648a40b0_0;
S_0000022a64872510 .scope task, "verify" "verify" 2 17, 2 17 0, S_0000022a6485ada0;
 .timescale -12 -12;
v0000022a648726a0_0 .var "a_task", 0 0;
v0000022a64872740_0 .var "b_task", 0 0;
v0000022a648727e0_0 .var "c_task", 0 0;
v0000022a64872880_0 .var "s_task", 0 0;
TD_half_adder_tb.verify ;
    %load/vec4 v0000022a648726a0_0;
    %store/vec4 v0000022a64872920_0, 0, 1;
    %load/vec4 v0000022a64872740_0;
    %store/vec4 v0000022a648a40b0_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000022a64872880_0;
    %load/vec4 v0000022a648a41f0_0;
    %cmp/e;
    %flag_get/vec4 4;
    %jmp/0 T_0.2, 4;
    %load/vec4 v0000022a648727e0_0;
    %load/vec4 v0000022a648a4150_0;
    %cmp/e;
    %flag_get/vec4 4;
    %and;
T_0.2;
    %flag_set/vec4 8;
    %jmp/0xz  T_0.0, 8;
    %vpi_call 2 28 "$display", "CORRECTO: Dato de entrada a: %b, Dato de entrada b: %b, Dato de salida s esperado: %b, Dato de salida s obtenido: %b, Dato de salida c esperado: %b, Dato de salida c obtenido: %b", v0000022a64872920_0, v0000022a648a40b0_0, v0000022a64872880_0, v0000022a648a41f0_0, v0000022a648727e0_0, v0000022a648a4150_0 {0 0 0};
    %jmp T_0.1;
T_0.0 ;
    %vpi_call 2 30 "$display", "INCORRECTO: Dato de entrada a: %b, Dato de entrada b: %b, Dato de salida s esperado: %b, Dato de salida s obtenido: %b, Dato de salida c esperado: %b, Dato de salida c obtenido: %b", v0000022a64872920_0, v0000022a648a40b0_0, v0000022a64872880_0, v0000022a648a41f0_0, v0000022a648727e0_0, v0000022a648a4150_0 {0 0 0};
T_0.1 ;
    %end;
    .scope S_0000022a6485ada0;
T_1 ;
    %vpi_call 2 38 "$dumpfile", "half_adder_tb.vcd" {0 0 0};
    %vpi_call 2 39 "$dumpvars", 32'sb00000000000000000000000000000000, S_0000022a6485ada0 {0 0 0};
    %end;
    .thread T_1;
    .scope S_0000022a6485ada0;
T_2 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022a648726a0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022a64872740_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022a64872880_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022a648727e0_0, 0, 1;
    %fork TD_half_adder_tb.verify, S_0000022a64872510;
    %join;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022a648726a0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022a64872740_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022a64872880_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022a648727e0_0, 0, 1;
    %fork TD_half_adder_tb.verify, S_0000022a64872510;
    %join;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022a648726a0_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022a64872740_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022a64872880_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022a648727e0_0, 0, 1;
    %fork TD_half_adder_tb.verify, S_0000022a64872510;
    %join;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022a648726a0_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022a64872740_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022a64872880_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022a648727e0_0, 0, 1;
    %fork TD_half_adder_tb.verify, S_0000022a64872510;
    %join;
    %end;
    .thread T_2;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "half_adder_tb.v";
    "./../../src/verilog/half_adder.v";
